### DISCUSSION TITLE
似乎性能对内存十分敏锐

### DISCUSSION THREAD

**Discussion by 2ji3150** [Blinue/Magpie#801] [Dev Channel]
您好，想请教一下。

测了一下anime4k的frametime发现到magpie的性能似乎对内存速度十分的敏锐。
DDR5-4800和DDR5-6000的性能差异到了12.5%以上。（均双通道） 
> 例: renderTime从8ms变成7ms

item|contents
-|-
cpu|Ryzen 7600x
gpu|GeForce RTX 4070
ram|ddr5 6000 16GB x 2 (EXPO)
os|windows11 22h3
mapie版本|0.10.4
捕捉方式|Desktop Duplication
垂直同步|ON + 允许延迟
显示器|4k 144hz

进一步分析了渲染时间，每一个步骤，包括bicubic的处理时间都有显著差异
由于DDR5-4800应该已经很快了，还有这么大的差异几乎只有在压缩软件等需要大量使用内存的才看得到
这是正常的吗，是什么原因导致这个软件对内存这么敏感呢。

```
由於不懂内部处理，我目前只能大胆猜测是目前设计上大量使用内存和vram之间搬动。
例如放大2次（4倍），没有办法再vram内一次完结。
又或是桌面捕捉性能和内存速度有关联（？）
但由于cpu并不弱，所以造成这样的结果我很惊讶
```

也欢迎其他网友测试，我很好奇再上去DDR5-7200或是搭载大量L3缓存的X3D CPU会不会再进一步影响性能。

<hr>

另外，我观察到一件事，

在我的环境，1080p超分4k的情况下，renderTime是9ms不变，不知道为什么frameTime确能维持6.9ms （144hz）。
但在720p超分4k的情况下，renderTime是7ms不变，frameTime却有时候会在动态的时候一瞬间往下掉，（例如144hz下，瞬间出现140hz）

由于720p超分4k相较1080p超分4k处理步骤多了2个（放大2次）所以会出现内存搬动的延迟吗？
我的理解，renderTime要小于frameTime才不会掉帧，有一点看不明白。

**Comment by cqaqlxz** [comment]
你是否使用的是集成显卡来运行 Magpie？集显的性能部分取决于内存。

**Reply by 2ji3150** [reply]
谢谢回答，否，Nvdia RTX4070。我更新了更详细的设置讯息。

**Comment by Blinue** [comment]
性能问题很复杂，我想可能有两个因素：

1. Desktop Duplication 可能会在内存和显存之间传输数据，如果帧数据可以更快地被加载到GPU，性能就会提高。
2. 高性能内存可能支持某些更先进的技术和特性，有利于显卡驱动优化性能。

内存性能肯定对帧率有影响，Magpie 在渲染时没有涉及到内存和显存的数据传输，但捕获时就不一定了。你可以试试其他其他捕获方式。

> 在我的环境，1080p超分4k的情况下，renderTime是9ms不变，不知道为什么frameTime确能维持6.9ms （144hz）。

这是因为你启用了“允许额外的延迟”，这会启用并行渲染，即前一帧还未渲染完成就会开始渲染下一帧了，提高了 GPU 的利用率，因此能提高性能。

**Reply by 2ji3150** [reply]
感谢回答。
想用新渲染系统测试看看，
但是新渲染新系统的build的游戏内叠加层和显示fps似乎不能作用。

**Reply by Blinue** [reply]
新渲染系统还有很多功能没有实现，请耐心等待

**Comment by 2ji3150** [comment]
测试了最新版，在任何模式下，内存的频率都显著影响处理性能(ddr5-4800 vs ddr5-6000)，真是奇怪的现象。希望有机会收集大家的 benchmark，来研究一下如何最小化延迟。

**Reply by 2ji3150** [reply]
最让我好奇的是搭载 X3D 大 L3 缓存的 CPU，或者 GDDR7 是否会带来显著提升。反过来说，或许当帧率达到某个点时，升级 GPU 的效果可能并不显著（？）
